packages/better-auth/src/plugins/organization/routes/crud-invites.ts

```diff
@@ -4,115 +4,119 @@ import { getSessionFromCtx } from "../../../api/routes";
+					message: "Organization not found",
+				});
+			}
+			const adapter = getOrgAdapter(ctx.context, ctx.context.orgOptions);
+			const member = await adapter.findMemberByOrgId({
+				userId: session.user.id,
+				organizationId: organizationId,
 			});
-		}
-		const alreadyMember = await adapter.findMemberByEmail({
-			email: ctx.body.email,
-			organizationId: organizationId,
-		});
-		if (alreadyMember) {
-			throw new APIError("BAD_REQUEST", {
-				message: "User is already a member of this organization",
+			if (!member) {
+				throw new APIError("BAD_REQUEST", {
+					message: "Member not found!",
+				});
+			}
+			const role = ctx.context.roles[member.role];
+			if (!role) {
+				throw new APIError("BAD_REQUEST", {
+					message: "Role not found!",
+				});
+			}
+			const canInvite = role.authorize({
+				invitation: ["create"],
 			});
-		}
-		const alreadyInvited = await adapter.findPendingInvitation({
-			email: ctx.body.email,
-			organizationId: organizationId,
-		});
-		if (alreadyInvited.length && !ctx.body.resend) {
-			throw new APIError("BAD_REQUEST", {
-				message: "User is already invited to this organization",
+			if (canInvite.error) {
+				throw new APIError("FORBIDDEN", {
+					message: "You are not allowed to invite members",
+				});
+			}
+			const alreadyMember = await adapter.findMemberByEmail({
+				email: ctx.body.email,
+				organizationId: organizationId,
 			});
-		}
-		const invitation = await adapter.createInvitation({
-			invitation: {
-				role: ctx.body.role,
+			if (alreadyMember) {
+				throw new APIError("BAD_REQUEST", {
+					message: "User is already a member of this organization",
+				});
+			}
+			const alreadyInvited = await adapter.findPendingInvitation({
 				email: ctx.body.email,
 				organizationId: organizationId,
-			},
-			user: session.user,
-		});
+			});
+			if (alreadyInvited.length && !ctx.body.resend) {
+				throw new APIError("BAD_REQUEST", {
+					message: "User is already invited to this organization",
+				});
+			}
+			const invitation = await adapter.createInvitation({
+				invitation: {
+					role: ctx.body.role as string,
+					email: ctx.body.email,
+					organizationId: organizationId,
+				},
+				user: session.user,
+			});
 
-		const organization = await adapter.findOrganizationById(organizationId);
+			const organization = await adapter.findOrganizationById(organizationId);
 
-		if (!organization) {
-			throw new APIError("BAD_REQUEST", {
-				message: "Organization not found",
-			});
-		}
+			if (!organization) {
+				throw new APIError("BAD_REQUEST", {
+					message: "Organization not found",
+				});
+			}
 
-		await ctx.context.orgOptions.sendInvitationEmail?.(
-			{
-				id: invitation.id,
-				role: invitation.role,
-				email: invitation.email,
-				organization: organization,
-				inviter: {
-					...member,
-					user: session.user,
+			await ctx.context.orgOptions.sendInvitationEmail?.(
+				{
+					id: invitation.id,
+					role: invitation.role as string,
```

Comment:
```suggestion
					role: invitation.role,
```
`invitation.role` is already define as a string